godot-core/src/obj/raw_gd.rs

```diff
@@ -438,15 +438,13 @@ where
 
         let callbacks = crate::storage::nop_instance_callbacks();
         let token = sys::get_library() as *mut std::ffi::c_void;
-        let binding = interface_fn!(object_get_instance_binding)(self.obj_sys(), token, &callbacks);
+        let binding = interface_fn!(object_get_instance_binding)(self.obj_sys(), token, &callbacks)
+            as sys::GDExtensionClassInstancePtr;
```

Comment:
Please avoid `as` casts, they're the crowbar of the type system.
(I should probably look for clippy lints at some point...)

Use `<*mut T>::cast()` instead, maybe in a separate variable.